complete-firebase-authentication-react-tutorial.md

```diff
@@ -834,227 +1057,230 @@ class SignUpForm extends Component {
+It is almost the same as the sign up form. Its input fields capture all the necessary information like username and password. A validation step makes sure the email and password are set before performing the request by enabling or disabling the submit button. The authentication API is used again, this time with a function to sign in the user rather than sign them up. If sign-in succeeds, the local state is updated with the initial state and the user is redirected again. If the sign-in fails, an error object is stored in the local state and an error message appears. The SignUpLink, which was defined earlier in the SignUp module, is used on the sign in page. It lets users sign up if they don't have an account, and it is found on the sign in page.
 
-{{% chapter_header "Sign Out with React and Firebase" "react-firebase-sign-out" %}}
+### Exercises:
+
+* Familiarize yourself with the SignIn and SignInForm components.
+  * If they are mysterious to you, checkout the previous section with the implementation of the SignUpForm again
+* Confirm your {{% a_blank "source code for the last section" "https://github.com/the-road-to-react-with-firebase/react-firebase-authentication/commit/6cbccce21952e10472504fc6ef3d37af2c998bb6" %}}.
 
-In order to complete the authentication loop, let's implement as last part the sign out component. It doesn't need any form and is only a button which shows up in the Navigation component in the next steps. Since we can use the already defined authentication API to sign out a user, it is fairly straight forward to pass the functionality to a button in a React component.
+{{% chapter_header "Sign Out with React and Firebase" "react-firebase-sign-out" %}}
 
-In *src/components/SignOut.js* file:
+To complete the authentication loop, next we'll implement the sign out component. The component is just a button that appears within the Navigation component. Since we can use the previously-defined authentication API to sign out a user, passing functionality to a button in a React component is fairly straightforward. Implement the SignOut in the *src/components/SignOut/index.js* file:
```

Comment:
"previously-defined" -> "previously defined"

Reply:
The dash is correct grammar here.